om4labs/diags/generic_yz_annual_bias_1x1deg/generic_yz_annual_bias_1x1deg.py

```diff
@@ -0,0 +1,385 @@
+    )
+    parser.add_argument(
+        "-d",
+        "--depth",
+        type=float,
+        default=None,
+        required=False,
+        help="depth of field compared to obs",
+    )
+    parser.add_argument(
+        "-l",
+        "--label",
+        type=str,
+        default="",
+        required=False,
+        help="Label to add to the plot",
+    )
+    parser.add_argument(
+        "-s",
+        "--suptitle",
+        type=str,
+        default="",
+        required=False,
+        help="Super-title for experiment. \
+                              Default is to read from netCDF file",
+    )
+
+    parser.add_argument("-m", "--model", type=str, default=None, help="Model Class")
+
+    parser.add_argument(
+        "-o",
+        "--outdir",
+        type=str,
+        default=".",
+        required=False,
+        help="output directory for plots",
+    )
+    parser.add_argument(
+        "-F",
+        "--format",
+        type=str,
+        default="png",
+        required=False,
+        help="output format type",
+    )
+    parser.add_argument(
+        "-O",
+        "--obsfile",
+        type=str,
+        nargs="+",
+        required=False,
+        help="File(s) containing obs data to compare against",
+    )
+    parser.add_argument(
+        "-D",
+        "--dataset",
+        type=str,
+        required=False,
+        default="WOA13_annual_TS",
+        help="Name of the observational dataset, \
+              as provided in intake catalog",
+    )
+    parser.add_argument(
+        "-i",
+        "--interactive",
+        action="store_true",
+        help="Interactive mode displays plot to screen. Default is False",
+    )
+    parser.add_argument(
+        "--platform",
+        type=str,
+        required=False,
+        default="gfdl",
+        help="computing platform, default is gfdl",
+    )
+    parser.add_argument(
+        "-S",
+        "--style",
+        type=str,
+        required=False,
+        default="diff",
+        help="output plot style (diff/compare)",
+    )
+    parser.add_argument(
+        "--static", "--static", type=str, default=None, help="Path to static file"
+    )
+
+    if template is True:
+        return parser.parse_args(None).__dict__
+    else:
+        return parser.parse_args(cliargs)
+
+
+def run(dictArgs):
+    """main can be called from either command line and then use parser from run()
+    or DORA can build the args and run it directly"""
+
+    # read the data needed for plots
+    y, z, depth, area, code, model, obs = read(dictArgs)
+
+    if len(code.shape) == 2:
+        code = np.tile(code[None, :, :], (len(z), 1, 1))
+
+    figs = []
+    filename = []
+
+    # global
+    figs.append(
+        plot(y, z, depth, area, model.mean(axis=-1), obs.mean(axis=-1), dictArgs)[0]
+    )
+    filename.append(
+        f"{dictArgs['outdir']}/{dictArgs['var']}_yz_gbl_{dictArgs['style']}"
+    )
+
+    # atlantic
```

Comment:
I think this code has to go either in compute or in plot. It breaks the modularity logic of om4labs.

Reply:
Good point - I'll move it into the plotting routine.

Reply:
See fb30f2d